sys/kern/subr_intr.c

```diff
@@ -800,8 +800,9 @@ pic_create(device_t dev, intptr_t xref, u_int flags)
 	struct intr_pic *pic;
 
 	mtx_lock(&pic_list_lock);
-	pic = pic_lookup_locked(dev, xref, flags);
+	pic = pic_lookup_locked(dev, xref, flags & ~FLAG_TYPE_MASK);
 	if (pic != NULL) {
+		pic->pic_flags |= flags & FLAG_TYPE_MASK;
```

Comment:
Which PIC driver requires this? I can't see any uses of intr_msi_register() in Xen.

Reply:
This was simply trying to preserve behavior of the `pic_init_secondary()` hooks.  By merging them controllers only have this called _once_.

Reply:
"Trying" meaning that it's supposed to work but isn't tested? It's not clear to me.

Reply:
Meaning the case I have the GICv3 doesn't have MSI capability, so this isn't well-tested.  _That_ is where this comes into play.

Reply:
Similar to the above #1280 and #1363.  I'm still a bit unsure of how to handle this though.